pints/_mcmc/__init__.py

```diff
@@ -301,24 +303,54 @@ def __init__(
             self, log_pdf, chains, x0, sigma0=None, transformation=None,
             method=None):
 
-        # Check function
-        if not isinstance(log_pdf, pints.LogPDF):
-            raise ValueError('Given function must extend pints.LogPDF')
+        if isinstance(log_pdf, pints.LogPDF):
+            self._multi_logpdf = False
+
+        else:
+            self._multi_logpdf = True
+            try:
+                log_pdfs_iter = iter(log_pdf)  # noqa
+            except TypeError:
+                raise ValueError('`log_pdf` must either extend pints.LogPDF, '
```

Comment:
I know it used to be a valueerror, but this should have been a typeerror

> Passing arguments of the wrong type (e.g. passing a [list](https://docs.python.org/3/library/stdtypes.html#list) when an [int](https://docs.python.org/3/library/functions.html#int) is expected) should result in a [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError), but passing arguments with the wrong value (e.g. a number outside expected boundaries) should result in a [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError).

https://docs.python.org/3/library/exceptions.html#TypeError

Reply:
Ok, makes sense. I've gone through the errors here and tried to be more careful with Value vs Type